packages/es-dev-server/src/middleware/babel.js

```diff
@@ -116,12 +118,24 @@ export function createBabelMiddleware(cfg) {
       ctx.body = compiled;
       return undefined;
     } catch (error) {
+      let errorMessage = error.message;
+
+      // replace babel error messages file path with the request url for readability
+      if (errorMessage.startsWith(filePath)) {
+        errorMessage = errorMessage.replace(filePath, ctx.url);
+      }
+
       // send compile error to browser for logging
-      ctx.body = `Error compiling: ${error.message}`;
+      ctx.body = `Error compiling: ${errorMessage}`;
       ctx.status = 500;
-      sendMessageToActiveBrowsers('error-message', JSON.stringify(error.message));
-      /* eslint-disable-next-line no-console */
-      console.error(error);
+
+      // strip babel ansi color codes because they're not colored correctly for the browser terminal
+      sendMessageToActiveBrowsers('error-message', JSON.stringify(stripAnsi(errorMessage)));
```

Comment:
can we convert rather than strip?

Reply:
I tried that but didnt have much success. We dont what color the user's terminal is so it is ttricky